### PR TITLE
Test techs checksum

### PR DESCRIPTION
**Pull request by o01eg**
Before I continue to convert techs to Python this PR adds check for techs chcksum

### CODE REVIEW

**Comment by geoffthemedio** [comment]
consistent with my checksums

File: test/system/TestChecksum.cpp [hunk 1]
```diff
@@ -4,10 +4,37 @@
...
+    }
+    if (force) {
+        BOOST_REQUIRE_MESSAGE(expected == value, env << " expected " << expected << " was " << value);
+    } else {
+        BOOST_WARN_MESSAGE(expected == value, env << " expected " << expected << " was " << value);
+    }
+}
+
+/**
+ * - Enforces named value reference checksum test if FO_CHECKSUM_NAMED_VALUEREF set.
```
Comment (geoffthemedio) [inline test/system/TestChecksum.cpp]:
```suggestion
 * - Enforces named value reference checksum test if FO_CHECKSUM_NAMED_VALUEREF is set.
```

File: test/system/TestChecksum.cpp [hunk 2]
```diff
@@ -4,10 +4,37 @@
...
+ * - Enforces techs checksum test if FO_CHECKSUM_TECH set.
```
Comment (geoffthemedio) [inline test/system/TestChecksum.cpp]:
```suggestion
 * - Enforces techs checksum test if FO_CHECKSUM_TECH is set.
```

File: test/system/TestChecksum.cpp [hunk 3]
```diff
@@ -4,10 +4,37 @@
...
+ * - If environment variable set to integer value that value test with calculated checksum.
```
Comment (geoffthemedio) [inline test/system/TestChecksum.cpp]:
```suggestion
 * - Setting each of these environment variables to an integer value will test that value against the corresponding parsed content checksum.
```

File: test/system/TestChecksum.cpp [hunk 4]
```diff
@@ -4,10 +4,37 @@
 
 BOOST_FIXTURE_TEST_SUITE(TestChecksum, ClientAppFixture)
 
+void TestCheckSumFromEnv(const char* env, unsigned int def, unsigned int expected) {
+    bool force = false;
+    unsigned int value = def;
+
+    const char *env_value = std::getenv(env);
+    if (env_value) {
```
Comment (geoffthemedio) [inline test/system/TestChecksum.cpp]:
```suggestion
    if (const char *env_value = std::getenv(env)) {
```

File: test/system/TestChecksum.cpp [hunk 5]
```diff
@@ -4,10 +4,37 @@
...
+        force = true;
```
Comment (geoffthemedio) [inline test/system/TestChecksum.cpp:12]: should `force` be set before the `try` and in the case that the `lexical_cast` throws?

File: test/system/TestChecksum.cpp [hunk 6]
```diff
@@ -4,10 +4,37 @@
 
 BOOST_FIXTURE_TEST_SUITE(TestChecksum, ClientAppFixture)
 
+void TestCheckSumFromEnv(const char* env, unsigned int def, unsigned int expected) {
```
Comment (geoffthemedio) [inline test/system/TestChecksum.cpp:7]: so `def` is a backup value to use for the check if the `getenv` lookup fails or the cast throws?

Comment (o01eg) [reply test/system/TestChecksum.cpp:7]: Yes, it is a default value.

File: test/system/TestChecksum.cpp (hunk 5 above)
Comment (o01eg) [reply test/system/TestChecksum.cpp:12]: No, in that case it should force check default value.

File: test/system/TestChecksum.cpp (hunk 3 above)
Comment (o01eg) [reply test/system/TestChecksum.cpp]: Fixed.

File: test/system/TestChecksum.cpp (hunk 2 above)
Comment (o01eg) [reply test/system/TestChecksum.cpp]: Fixed.

File: test/system/TestChecksum.cpp (hunk 1 above)
Comment (o01eg) [reply test/system/TestChecksum.cpp]: Fixed.

**Comment by geoffthemedio** [comment]
otherwise OK

**Pull request merged**